bitcoin/src/psbt/mod.rs

```diff
@@ -213,11 +213,14 @@ impl Psbt {
                 if fee_rate > max_fee_rate {
                     return Err(ExtractTxError::AbsurdFeeRate { fee_rate, tx });
                 }
+                Ok(tx)
             }
-            NumOpResult::Error(_) => unreachable!("weight() is always non-zero"),
+            // We hit this if fee * 4,000 overflows.
```

Comment:
In c77e7ce96c366f8cb1a1de7cfcbd8daf5980cca3:

We should fix division to not overflow. It's super surprising that division of an integer number of sats by an integer number of wu needs to do some sort of scaling factor shenanigans. And it's not even the most precise thing we could be doing.